toolbox/dicom.py

```diff
@@ -97,43 +107,75 @@ def get_elements(self, ds):
-        'Rows',
-        'Columns',
-        'Pixel Spacing',
-        'Pixel Representation',  # Pixel Representation (0028,0103) is either unsigned (0) or signed (1)
-        'Window Center',
-        'Window Width',
-        'Rescale Intercept',
-        'Rescale Slope',
-        'Rescale Type',
-    ]
-    keys = tag_of_interested + [SUBJECT_ID, SEQUENCE_NAME]
-
-    @classmethod
-    def get_df(cls, df_dcm_file):
-        dicom_tag_processor = DicomTagProcessor()
-        subject_list = []
-        for _, row in df_dcm_file.iterrows():
-            ds = pydicom.dcmread(row["path"], stop_before_pixels=True)
-            elements = dicom_tag_processor.get_elements(ds)
+    def get_tags_of_interest(self, dicom_file, tags_of_interest):
+        keys, subject_list = tags_of_interest + [SUBJECT_ID, SEQUENCE_NAME], []
+        for _, row in DicomTagProcessor._get_dicom_df(dicom_file).iterrows():
+            ds = pydicom.dcmread(row[DICOM_PATH], stop_before_pixels=True)
+            elements = self.get_elements(ds)
             data = {
-                "subject_id": row["subject_id"],
-                "sequence_name": row["sequence_name"],
+                SUBJECT_ID: row[SUBJECT_ID],
+                SEQUENCE_NAME: row[SEQUENCE_NAME],
             }
             data.update(elements)
             subject_list.append(data)
-        return pd.DataFrame([{k: v for k, v in s.items() if k in cls.keys} for s in subject_list])
+        return pd.DataFrame([{k: v for k, v in s.items() if k in keys} for s in subject_list])
+
+
+def get_file_number(path):
+    """ Return DICOM file number.
+    :param path: absolute path to DICOM file.
+    :return: slice number - integer
+    """
+    return int(re.findall(r'\d+', path)[0])
+
+
+def get_path_to_first_dicom_file(dicom_folder):
+    """ Return absolute path to first DICOM file (i.e. first slice)
+    :param dicom_folder: absolute path to directory which contains list of
+    subject specific DICOM files.
+    :return: absolute DICOM file path that corresponds to the first slice.
+    """
+    min_number_file = [99999999, None]
+    for dcm_file in glob.glob(os.path.join(dicom_folder, "*")):
+        file_number = get_file_number(dcm_file)
+        if file_number < min_number_file[0]:
+            min_number_file[0] = file_number
+            min_number_file[1] = dcm_file
+    return min_number_file[1]
+
+
+def get_path_to_middle_file(dicom_folder):
+    """ Return absolute path to middle DICOM file (i.e. middle slice)
+    :param dicom_folder: absolute path to directory which contains list of
+    subject specific DICOM files.
+    :return: absolute DICOM file path that corresponds to the middle slice.
+    """
+    slice_ids_dict = {}
+    for dcm in os.listdir(dicom_folder):
+        slice_id = get_file_number(dcm)
+        slice_ids_dict[slice_id] = os.path.join(dicom_folder, dcm)
+    slice_ids = list(slice_ids_dict.keys())
+    slice_ids.sort()
+    return slice_ids_dict[slice_ids[len(slice_ids)//2]]
```

Comment:
I remember sometimes this doesn't give you the right middle slice. But maybe we are fine here as we just want to pick a slice

Reply:
maybe natsort?

Reply:
I think we are fine, but I will check @johnccshen suggestion.